src/components/Lesson/lesson.helpers.ts

```diff
@@ -327,3 +331,68 @@ export function groupLessonPathways(pathways: LessonPathway[]) {
     subjects,
   };
 }
+
+type Attribution = {
+  questionNumber: string;
+  attribution: string;
+};
+
+export const createAttributionObject = (
+  questions: LessonOverviewQuizData,
+): Attribution[] | [] => {
+  if (questions) {
+    const attributions: Attribution[] = questions.reduce(
+      (acc: Attribution[], question, index) => {
+        const questionNumber = `Q${index + 1}`;
+        if (question && question.questionStem) {
+          const { questionStem } = question;
+          const imageStems = questionStem.filter(
+            (stem) =>
+              stem.type === "image" &&
+              !Array.isArray(stem.image_object.metadata) &&
+              stem.image_object.metadata.attribution &&
+              stem.image_object,
+          ) as StemImageObject[];
+          const mappedAttributions: Attribution[] = imageStems.map((stem) => {
+            if (
+              !Array.isArray(stem.image_object.metadata) &&
+              stem.image_object.metadata.attribution
+            ) {
+              return {
+                questionNumber,
+                attribution: stem.image_object.metadata.attribution,
+              };
+            }
+          }) as Attribution[];
+          acc.push(...mappedAttributions);
+        }
+        if (question && question.answers) {
+          const { answers } = question;
+          if (answers["multiple-choice"]) {
+            const { "multiple-choice": multipleChoice } = answers;
+            multipleChoice.forEach(({ answer }, index) => {
+              answer.forEach((stem) => {
+                if (
+                  stem.type === "image" &&
+                  !Array.isArray(stem.image_object.metadata) &&
+                  stem.image_object.metadata &&
+                  stem.image_object.metadata.attribution
+                ) {
+                  acc.push({
+                    questionNumber: `${questionNumber} image ${index + 1}`,
+                    attribution: stem.image_object.metadata.attribution,
+                  });
+                }
+              });
+            });
```

Comment:
I don't want to hold up this PR but it would be good to have a follow up to refactor this method a bit to make it more readable! breaking down into smaller sub methods and extracting the conditions for example

Reply:
Yeah agreed, is a bit long some of the additional checks in the function are because of TS not playing ball